src/platform/graphics/pixman_image_scaling.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "mir/graphics/pixman_image_scaling.h"
+
+#include "mir/graphics/cursor_image.h"
+#include "mir_toolkit/common.h"
+
+#include "pixman-1/pixman.h"
+
+#include <cmath>
+#include <cstring>
+
+mir::graphics::PixBuffer mir::graphics::scale_cursor_image(
+    mir::graphics::CursorImage const& cursor_image, float new_scale)
+{
+    auto const [width, height] = cursor_image.size();
+
+    // Special case: no resizing. Just copy the data.
+    if(new_scale == 1)
```

Comment:
Possibly clearer without an implicit conversion?

```suggestion
    if(new_scale == 1.0f)
```